### DISCUSSION TITLE
Add `volume` sensor class

### DISCUSSION THREAD

**Discussion by epenet** [home-assistant/architecture#804] [Entity Models] [answered]
## Proposal

Add a `sensor` `device_class` for `volume` to home assistant core and frontend (+ documentation).

To solve the problem with different applications using different units for volume, without the need to add device classes for each of those applications, the unit of a `volume` sensor shall be configurable by the user in the same way as for temperature and pressure sensors.

Specific device classes for common applications may of course be added if there is a need.

A draft PR can be found in https://github.com/home-assistant/core/pull/77960

**Comment by parautenbach** [comment]
Just as a point of clarity: This is spatial volume, not sound volume (as in sound pressure level), right?

**Reply by epenet** [reply]
Yes correct.
It would apply initially to liters, milliliters, gallons (US), fluid ounces (US), cubic feet and cubic meters.

**Comment by emontnemery** [comment, marked as answer]
We've discussed this topic and concluded that it's ok to add a general sensor device class `volume` as suggested.
